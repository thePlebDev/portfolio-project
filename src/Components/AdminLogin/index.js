import React,{useState} from 'react';
import styled from 'styled-components';
import Form from '../Form';

import useAdminForm from '../../Hooks/useAdminForm';
import validation from '../../utils/Validation';



const Container = styled.div`
  width:80%;
  border: 1px solid red;
  margin:0 auto;
`


const AdminLogin =()=>{
  const {state,handleChange,handleSubmit} = useAdminForm(validation.adminLogin)
  return(
    <Container>
      <Form onSubmit={handleSubmit}>
        <label>
          username
          <input name="username" value={state.username} onChange={(e)=>handleChange(e)}/>
        </label>

        <label>
        password
          <input name="password" value={state.password} onChange={(e)=>handleChange(e)}/>
        </label>
        <button type='Submit'> Submit</button>
      </Form>
    </Container>
  )
}


export default AdminLogin
