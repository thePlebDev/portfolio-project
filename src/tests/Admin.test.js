import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import AdminLogin from '../Components/AdminLogin'


//1)snapshot testing
//2) testing the props(default and custom)
//3) testing the data types
//4)data type prop testing.

describe('testin the admin login',()=>{
  it('should render properly',()=>{
    const tree = renderer.create(<AdminLogin />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
