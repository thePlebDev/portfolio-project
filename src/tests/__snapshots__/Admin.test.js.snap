// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testin the admin login should render properly 1`] = `
<div
  className="sc-gKAblj fRSZFp"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    style={
      Object {
        "fontSize": "80px",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
    />
  </svg>
  <form
    className="sc-bdnylx kFpyQD"
    onSubmit={[Function]}
  >
    <div
      className="sc-hKFyIo ierqhF"
    >
      <label
        for="username"
      >
        <input
          className="sc-eCApGN gWxIsj"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="username"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="sc-gtssRu fSclDs"
    >
      <label
        id="password"
      >
        <input
          className="sc-dlnjPT lhztcp"
          for="password"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
      </label>
    </div>
    <button
      className="sc-jSFkmK FOhJR"
      type="submit"
    >
      SEND
    </button>
  </form>
</div>
`;
